ext/POSIX/lib/POSIX.pm

```diff
@@ -247,6 +246,18 @@ sub sprintf {
     CORE::sprintf(shift,@_);
 }
 
+my @_atexit;
+
+END {
+    my $f;
+    $f->() while $f = pop @_atexit;
+}
+
+sub atexit {
+    usage "atexit(sub { ... })" if @_ != 1;
+    push @_atexit, $_[0];
```

Comment:
We could get a type check (sort of) here by doing
```perl
push @_atexit, \&{$_[0]};
```
So is `$_[0]` is not a coderef, an error is thrown here.

Reply:
With this:

```perl
push @_atexit, \&{$_[0]};
```

... then doing:

```
atexit("str")
```

Would give the following message:

```
t/attext.t .......... 1/4 str is not a valid POSIX macro at /home/gugod/Projects/Perl/perl5/ext/POSIX/../../lib/POSIX.pm line 252
END failed--call queue aborted.
```

Which abort the program, but then it's  not directly helpful.

Since this PR would be based on a perl > 5.40...perhaps a brief check with `builtin::reftype` would be better ?

Reply:
I ended up doing:

```perl
    usage "atexit(sub { ... })"
        unless @_ == 1 && defined($_[0]) && (reftype($_[0]) // '') eq 'CODE';
```

... after importing `reftype` from `bulitin`. I assumed that importing `builtin` is OK since this PR is already based on 5.41.

Reply:
For efficiency, I would skip `use builtin` and just call `builtin::reftype($_[0])` directly. (`use builtin` doesn't do much but locate and run an almost empty `.pm` file. The functions themselves are always defined in the perl interpreter itself.)

That said, there is one corner case that this doesn't handle: Objects with an overloaded `&{}` operator. The condition for those would be:
```perl
use overload ();
...

    ... unless
        @_ == 1 && (
            (builtin::reftype($_[0]) // '') eq 'CODE')
            || (builtin::blessed($_[0]) && overload::Method($_[0], '&{}'))
        );
```